File: VolleyballAPI/VolleyballAPI/Entities/Tournament.cs
using System.ComponentModel.DataAnnotations;

namespace VolleyballAPI.Entities
{
  public class Tournament
  {
    [Required]
    [Key]
    public Guid Id { get; set; }
    public Guid LocationId { get; set; }
    [Required]
    [MaxLength(100), MinLength(5)]
    public string Name { get; set; }
    public DateTime Date { get; set; }
    public DateTime EntryDeadline { get; set; }
    public string Organizer { get; set; }
    public string RegistrationPolicy { get; set; }
    public string TeamPolicy { get; set; }
    public int Courts{ get; set; }
    public List<int> MaxTeamsPerLevel{ get; set; }
    public string Description { get; set; }
    public string PictureLink { get; set; }
    public Level Categories { get; set; }
    public PriceType PriceType { get; set; }
    public TournamentType TournamentType{ get; set; }


    public virtual Location Location { get; set; }
    public virtual ICollection<TournamentCompetitor> Teams { get; set; }
    public virtual ICollection<Match> Matches { get; set; }
    public virtual ICollection<FavouriteTournament> UserHasAsFavourite { get; set; }
    }
}

File: VolleyballManagementAppFrontend/src/app/models/entities/tournament.ts
import { Level } from '../enums/level';
import { Location } from './location';
import { Match } from './match';
import { PriceType } from '../enums/priceType';
import { Team } from './team';
import { TournamentType } from '../enums/tournamentType';
import { User } from './user';

export interface Tournament {
  id: string;
  name: string;
  date: Date;
  entryDeadline: Date;
  description: string;
  organizer: string;
  registrationPolicy: string;
  teamPolicy: string;
  pictureLink: string;
  categories: Level[];
  priceType: PriceType;
  teams: Team[];
  matches: Match[];
  userHasAsFavourite: User[];
  location: Location;
  courts: number;
  maxTeamsPerLevel: number[];
  tournamentType: TournamentType;
}
 